go/service/user.go

```diff
@@ -344,6 +345,248 @@ func (h *UserHandler) UploadUserAvatar(ctx context.Context, arg keybase1.UploadU
+	if err != nil {
+		return ret, err
+	}
+	for _, suggestion := range suggestions {
+		ret = append(ret, keybase1.ProofSuggestion{
+			Key:     suggestion.Key,
+			Text:    suggestion.PickerText,
+			Subtext: suggestion.PickerSubtext,
+			Icon:    suggestion.PickerIcon,
+		})
+	}
+	return ret, nil
+}
+
+type ProofSuggestion struct {
+	Key           string
+	ProfileText   string                // "Prove your Twitter", "Add a PGP key"
+	ProfileIcon   []keybase1.SizedImage // xxx fill these in
+	PickerText    string                // "Twitter", "Your own website", "octodon.xyz"
+	PickerSubtext string                // "twitter.com", "Mastodon instance"
+	PickerIcon    []keybase1.SizedImage // xxx fill these in
+	Priority      int
+}
+
+func (h *UserHandler) profileProofSuggestionsHelper(mctx libkb.MetaContext) (ret []ProofSuggestion, err error) {
+	user, err := libkb.LoadMe(libkb.NewLoadUserArgWithMetaContext(mctx).WithPublicKeyOptional())
+	if err != nil {
+		return ret, err
+	}
+	if user == nil {
+		return ret, fmt.Errorf("could not load logged-in user")
+	}
+
+	var dummyIcon []keybase1.SizedImage // TODO CORE-9882: Get some icons
+	var suggestions []ProofSuggestion
+	ossifiedSocial := map[string]ProofSuggestion{
+		"github": {
+			Key:           "github",
+			ProfileText:   "Prove your GitHub",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "GitHub",
+			PickerSubtext: "github.com",
+			PickerIcon:    dummyIcon,
+		},
+		"hackernews": {
+			Key:           "hackernews",
+			ProfileText:   "Prove your Hacker News",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Hacker News",
+			PickerSubtext: "news.ycombinator.com",
+			PickerIcon:    dummyIcon,
+		},
+		"reddit": {
+			Key:           "reddit",
+			ProfileText:   "Prove your Reddit",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Reddit",
+			PickerSubtext: "reddit.com",
+			PickerIcon:    dummyIcon,
+		},
+		"twitter": {
+			Key:           "twitter",
+			ProfileText:   "Prove your Twitter",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Twitter",
+			PickerSubtext: "twitter.com",
+			PickerIcon:    dummyIcon,
+		},
+		"rooter": {
+			Key:           "rooter",
+			ProfileText:   "Prove your Rooter",
+			ProfileIcon:   dummyIcon,
+			PickerText:    "Rooter",
+			PickerSubtext: "",
+			PickerIcon:    dummyIcon,
+		}}
+	serviceKeys := mctx.G().GetProofServices().ListServicesThatAcceptNewProofs()
+	for _, service := range serviceKeys {
+		switch service {
+		case "web", "dns", "http", "https":
+			// These are under the "web" umbrella.
+			// "web" is added below.
+			continue
+		}
+		serviceType := mctx.G().GetProofServices().GetServiceType(service)
+		if serviceType == nil {
+			mctx.CDebugf("missing proof service type: %v", service)
+			continue
+		}
+		if len(user.IDTable().GetActiveProofsFor(serviceType)) > 0 {
+			mctx.CDebugf("user has an active proof: %v", serviceType.Key())
+			continue
+		}
+		if suggestion, ok := ossifiedSocial[service]; ok {
```

Comment:
won't we always use the hardcoded vals above then?

Reply:
I'm not sure what you're asking. Yes the key "twitter" will always say "Prove your Twitter" even if the server sends something else.

Reply:
why don't we want it to be configurable?

Reply:
3 weak reasons. Less wiggle room for a malicious server. This is easy. Why would we?

Reply:
- if paramproofs are under server control we should just have a single point of truth, i don't think we gain anything for having some hard coded and some not.
- it's easy but also adds complication to debugging/extending in the future.
- aside from the pvl aspect of old proofs it would be nice if these were finally unified and behaving essentially the same for the other aspects of the system